src/main/java/com/google/sps/servlets/ProfileRoutes.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.Route;
+import com.google.sps.data.User;
+import java.io.IOException;
+import java.util.*;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@SuppressWarnings("serial")
+@WebServlet("/user-routes")
+public class ProfileRoutes extends HttpServlet {
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
There seems to be multiple copies of this function. It would be a good idea to move it to a helper class that can be referenced from all servlets. You can either to it in this PR, or add an issue and TODO to do it in a follow-up PR, whichever you prefer.

Reply:
Oh.... I just forgot to delete it, sorry.